numpy/core/src/multiarray/dtype_transfer.c

```diff
@@ -3462,13 +3464,14 @@ PyArray_GetDTypeTransferFunction(int aligned,
     }
 
     /*
-     * If there are no references and the data types are equivalent,
+     * If there are no references and the data types are equivalent and builtin,
      * return a simple copy
      */
     if (PyArray_EquivTypes(src_dtype, dst_dtype) &&
             !PyDataType_REFCHK(src_dtype) && !PyDataType_REFCHK(dst_dtype) &&
             ( !PyDataType_HASFIELDS(dst_dtype) ||
-              is_dtype_struct_simple_unaligned_layout(dst_dtype)) ) {
+              is_dtype_struct_simple_unaligned_layout(dst_dtype)) &&
```

Comment:
Indentation needs fix.

Reply:
I had meant to have this align with col 13 since it's part of the conjunction `&&` operands, so that it should align with the beginning of the parentheses (and `!PyDataType_REFCHK(...`l).

If you want, I can move this line up to make the continuation more evident?

Reply:
NVM, I misread the code.